### PR TITLE
Add ostype parameter in LXC container clone of Proxmox VE

### PR DESCRIPTION
**Pull request by blacknon**
##### SUMMARY

When creating an LXC container for Proxmox VE, containers such as openwrt could not be created because the ostype parameter could not be specified.
To deal with this, we added some notes so that the ostype parameter can be handled by Ansible.

##### ISSUE TYPE

- Docs Pull Request
- Feature Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/proxmox.py [hunk 1]
```diff
@@ -98,6 +98,12 @@
       - target storage
     type: str
     default: 'local'
+  ostype:
+    description:
+      - specifies the ostype of the lxc container
+    choices: ['auto', 'debian', 'devuan', 'ubuntu', 'centos', 'fedora', 'opensuse', 'archlinux', 'alpine', 'gentoo', 'nixos', 'unmanaged']
+    type: str
+    default: 'auto'
```
Comment (felixfontein) [inline plugins/modules/proxmox.py:106]:
```suggestion
    default: 'auto'
    version_added: 8.1.0
```

File: plugins/modules/proxmox.py [hunk 2]
```diff
@@ -497,6 +503,10 @@ def create_instance(self, vmid, node, disk, storage, cpus, memory, swap, timeout
                     self.module.fail_json(msg='%s is not a valid tag' % tag)
             kwargs['tags'] = ",".join(kwargs['tags'])
 
+        if 'ostype' in kwargs:
+            if kwargs['ostype'] == 'auto':
+                kwargs.pop('ostype')
```
Comment (felixfontein) [inline plugins/modules/proxmox.py:508]: Why not:
```suggestion
        if kwargs.get('ostype') == 'auto':
            kwargs.pop('ostype')
```

File: plugins/modules/proxmox.py [hunk 3]
```diff
@@ -98,6 +98,12 @@
       - target storage
     type: str
     default: 'local'
+  ostype:
+    description:
+      - specifies the ostype of the lxc container
```
Comment (felixfontein) [inline plugins/modules/proxmox.py:103]:
```suggestion
      - Specifies the C(ostype) of the LXC container.
      - If set to V(auto), no C(ostype) will be provided on instance creation.
```

**Review by felixfontein — COMMENTED** [review]
Thanks for your contribution! Please add a [changelog fragment](https://docs.ansible.com/ansible/devel/community/collection_development_process.html#creating-a-changelog-fragment). Thanks.

I've added some first comments.

File: plugins/modules/proxmox.py (hunk 2 above)
Comment (blacknon) [reply plugins/modules/proxmox.py:508]: Sorry, the code I wrote out of habit remained 🙇

**Comment by blacknon** [comment]
@felixfontein 
Thank you for the review!
Fixed it.

File: changelogs/fragments/7462-Add-ostype-parameter-in-LXC-container-clone-of-ProxmoxVE.yaml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+minor_changes:
+  - proxmox_ostype - now possible to specify the ostype when creating an lxc container. (https://github.com/ansible-collections/community.general/pull/7462).
```
Comment (felixfontein) [inline changelogs/fragments/7462-Add-ostype-parameter-in-LXC-container-clone-of-ProxmoxVE.yaml:2]:
```suggestion
  - proxmox_ostype - it is now possible to specify the ``ostype`` when creating an LXC container (https://github.com/ansible-collections/community.general/pull/7462).
```

**Comment by blacknon** [comment]
@felixfontein 
Thank you for the review!
I also fixed this.

**Review by felixfontein — APPROVED** [review]
Looks correct to me (though I don't know proxmox :) ). I will merge in ~a week if nobody objects.

**Comment by felixfontein** [comment]
@blacknon thanks for improving the modules!

**Pull request merged**